### ISSUE TITLE
oneOf, allOf, oneOf with references returning as invalid when should be valid

### ISSUE THREAD

**Issue by oliverslade** [JamesNK/Newtonsoft.Json.Schema#132]
Using https://www.jsonschemavalidator.net/ in draft 04

I tried the payload of:
```
{
    "foobar" : "foo",
    "baz" : "baz"
}
```
Validating against the schema:

```
{
    "oneOf": [
      {
        "allOf": [
          {
              "foobar": {
                "type": "string",
                "enum": ["foo"]
              }
          },
          { "$ref": "#/definitions/bazquux" }
        ]
      },
      {
        "allOf": [
          {
              "foobar": {
                "type": "string",
                "enum": ["bar"]
              }
          },
          { "$ref": "#/definitions/bazquux" }
        ]
      }
    ],
    "definitions": {
      "bazquux": {
        "oneOf": [
          {
            "properties": {
              "baz": {
                "type": "string"
              }
            },
            "required": ["baz"]
          },
          {
            "properties": {
              "quux": {
                "type": "string"
              }
            },
            "required": ["quux"]
          }
        ]
      }
    }
  }
```
This throws incorrect errors suggesting that the JSON does not match the schema when it does and can be shown in other schema validators such as https://jsonschemalint.com/#/version/draft-04/markup/json. Would it be possible to look into why that is the case?

**Comment by JamesNK** [comment]
You're missing `properties` here:

```
          {
              "foobar": {
                "type": "string",
                "enum": ["foo"]
              }
          }
```
